fibsem/microscopes/simulator.py

```diff
@@ -311,6 +332,89 @@ def acquire_image(self, image_settings: ImageSettings) -> FibsemImage:
 
         return image
 
+    def _generate_next_image(
+        self,
+        beam_type: BeamType,
+        output_shape: Tuple[int, int],
+        dtype: np.dtype = np.uint8,
+    ) -> np.ndarray:
+        """Generate the next image in the sequence for the specified beam type, 
+            formatted for the current acquisition settings.
+        Args:
+            beam_type: The type of beam (electron or ion).
+            output_shape: The shape of the output image (height, width).
+            dtype: The data type of the output image.
+        Returns:
+            np.ndarray: The generated image data.
+        """
+        try:
+            # get the next filename from the imaging system
+            filename = next(self.imaging_system.generator[beam_type])
```

Comment:
This could cause a key error if not there isn't a generator of that type, so this might be neater:
```
beam_generator = self.imaging_system.generator.get(beam_type)
if beam_generator is None:
    raise ValueError(f"No generator available for {beam_type}")
filename = next(beam_generator)
```